optimizer_wrapper.rb

```diff
@@ -729,27 +751,30 @@ def self.route_details(vrp, route, vehicle)
     details
   end
 
+  def self.compute_route_distances(vrp, route, vehicle)
+    details = route_details(vrp, route, vehicle)
+
+    return details unless details.to_a.size.positive?
+
+    route[:total_distance] = details.map(&:first).compact.reduce(:+).round
+    route[:activities][1..-1].each_with_index{ |activity, index|
+      activity[:travel_distance] = details[index].first if details[index]
+    }
+  end
+
   def self.parse_result(vrp, result)
     tic_parse_result = Time.now
     result[:routes].each{ |r|
       details = nil
-      v = vrp.vehicles.find{ |vehicle| vehicle.id == r[:vehicle_id] }
-      if r[:end_time] && r[:start_time]
-        r[:total_time] = r[:end_time] - r[:start_time]
-      end
+      unify_field_units(r)
```

Comment:
We need to calculate the totals first and then do the rounding of individual routes and stops of these routes. Otherwise, the rounding errors will accumulate.

Reply:
I have found my comment  @fonsecadeline.  You gave a thumbs up and resolved conversation so I assume you already did the modifications. Lets verify it next week as you suggested.

Reply:
It is okay now :)